axelrod/strategies/grudger.py

```diff
@@ -293,16 +294,18 @@ def strategy(self, opponent: Player) -> Action:
         The punishment is in the form of 'd' defections followed by a penance of
         'c' consecutive cooperations.
         """
```

Comment:
sorry.  that wasn't clear.  

are n=0, d=0 and c=0 legal inputs?  

if they are, the original code failed on n=0 and d=0.  This change is to fix that.  

if they are not allowed as inputs, then the original code works just fine and i can remove this change.  (but perhaps it would be useful to add to `n>=1 and d>=1` to the docstring)